business/iter8.go

```diff
@@ -0,0 +1,279 @@
+		// return models.ExperimentList{}, err
+	}
+
+	cacheToken := ""
+	kConfig := kialiConfig.Get()
+	if kConfig.InCluster {
+		if saToken, err := kubernetes.GetKialiToken(); err != nil {
+			return experiments, err
+			// return models.ExperimentList{}, err
+		} else {
+			cacheToken = saToken
+		}
+	}
+	log.Infof("Found token len %d", len(cacheToken))
+		k8sConfig.QPS = config.Get().KubernetesConfig.QPS
+		k8sConfig.Burst = config.Get().KubernetesConfig.Burst
+		k8sConfig.BearerToken = cacheToken
+		k8s, err := kube.NewForConfig(k8sConfig)
+
+	var timeout time.Duration
+	path := fmt.Sprintf("/apis/iter8.tools/v1alpha1/")
+	err = k8s.RESTClient().Get().AbsPath(path).
+		Namespace(namespace).
+		Resource("experiments").
+		Timeout(timeout).
+		Do().
+		Into(result)
+	log.Infof("Get Experiments return error %s", err)
+	log.Infof("Finding experiment for namespace %s", namespace)
+	log.Infof("Received result lenght %d", len(result.Items))
+	for _, item := range result.Items {
+
+		experiments  = append (experiments,  models.ExperimentListItem{
+			Name: item.Name,
+			Phase: string(item.Status.Phase),
+			Status: item.Status.Message,
+			CreatedAt: formatTime(item.CreationTimestamp.Time),
+			Baseline: item.Spec.TargetService.Baseline,
+			BaselinePercentage: item.Status.TrafficSplit.Baseline,
+			Candidate: item.Spec.TargetService.Candidate,
+			CandidatePercentage: item.Status.TrafficSplit.Candidate,
+			Namespace: namespace,
+		})
+	}
+	return experiments, nil
+}
+
+func (in *Iter8Service) Iter8ExperimentCreate(body []byte) ( *iter8v1alpha1.Experiment, error){
+	result := &iter8v1alpha1.Experiment{}
+	k8sConfig, err := kubernetes.ConfigClient()
+	if err != nil {
+		return nil, err
+		// return models.ExperimentList{}, err
+	}
+
+	cacheToken := ""
+	kConfig := kialiConfig.Get()
+	if kConfig.InCluster {
+		if saToken, err := kubernetes.GetKialiToken(); err != nil {
+			return nil,  err
+			// return models.ExperimentList{}, err
+		} else {
+			cacheToken = saToken
+		}
+	}
+	log.Infof("Found token len %d", len(cacheToken))
+	k8sConfig.QPS = config.Get().KubernetesConfig.QPS
+	k8sConfig.Burst = config.Get().KubernetesConfig.Burst
+	k8sConfig.BearerToken = cacheToken
+	k8s, err := kube.NewForConfig(k8sConfig)
+	newexperimentSpec := &models.ExperimentSpec{}
+	err2 := json.Unmarshal(body, newexperimentSpec)
+	if err2 != nil {
+		log.Errorf("JSON: %s shows error: %s", string(body), err2)
+		err := fmt.Errorf("Bad Experiment json")
+		return nil, err
+	}
+	experiment := iter8v1alpha1.Experiment {
+		TypeMeta: metav1.TypeMeta {
+			APIVersion: iter8v1alpha1.SchemeGroupVersion.String(),
+			Kind: "Experiment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      newexperimentSpec.Name,
+			Namespace: newexperimentSpec.Namespace,
+		},
+		Spec: iter8v1alpha1.ExperimentSpec {
+			TargetService: iter8v1alpha1.TargetService {
+				ObjectReference: &corev1.ObjectReference {
+					Name: newexperimentSpec.Service,
+					Namespace: "bookinfo-iter8",
```

Comment:
I guess these constants were left for a demo scenario.

Reply:
Oops., sorry about that. Thank you for catching that up.